internal/db/db.go

```diff
@@ -0,0 +1,66 @@
+package db
+
+import (
+	"database/sql"
+	"errors"
+	"time"
+
+	_ "github.com/lib/pq"
+)
+
+// Connect makes a connection to the PostgreSQL database
+// and returns the sql.DB handler representing a pool of zero or more
+// underlying connections.
+func Connect(connString string, maxConnections int) (*sql.DB, error) {
+	db, err := sql.Open("postgres", connString)
+	db.SetMaxOpenConns(maxConnections)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Try to connect up to 5 times
+	for retry := 1; retry <= 5; retry++ {
+		err = db.Ping()
+		if err == nil {
+			return db, nil
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	return nil, err
+}
+
+// Setup uses the existing database connection to create
+// the necessary tables for the API (if they don't exist).
+func Setup(db *sql.DB) error {
+	if db == nil {
+		return errors.New("cannot setup database, must call Connect() first")
+	}
+
+	if _, err := db.Exec(
+		`
+		CREATE TABLE IF NOT EXISTS app (
+	    	id uuid NOT NULL UNIQUE,
```

Comment:
Minor indentation change here

Reply:
`gofmt -w internal/db/db.go` should take care of that :+1: